build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchBuildCompletePlugin.java

```diff
@@ -159,7 +162,16 @@ public void execute(BuildFinishedFlowAction.Parameters parameters) throws FileNo
                     } catch (IOException e) {
                         processOutput = "";
                     }
-                    String artifactUuid = processOutput.trim();
+
+                    List<String> artifacts = Arrays.stream(processOutput.trim().split("\n")).map(String::trim).sorted((a, b) -> {
+                        String[] partsA = a.split(" ");
+                        String[] partsB = b.split(" ");
+                        // ISO-8601 timestamps can be sorted lexicographically
+                        return partsA[1].compareTo(partsB[1]);
+                    }).toList();
```

Comment:
I think we can simplify this somewhat by sorting in reverse order and calling `findFirst()` instead of `toList()` since we only care about the most recent one.